src/jupyterlabcontroller/storage/k8s.py

```diff
@@ -5,13 +5,15 @@
 import asyncio
 from base64 import b64encode
 from collections.abc import AsyncGenerator
-from typing import Optional
+from threading import Lock
```

Comment:
This is async code so you can't use `threading.Lock`. Best case is that it won't do anything; worst case is that it will deadlock the process. You have to use `asyncio.Lock`.

Reply:
Oooh, thank you.

Reply:
Done